lib/modules/manager/helmfile/artifacts.ts

```diff
@@ -59,9 +69,42 @@ export async function updateArtifacts({
         constraint: config.constraints?.kustomize,
       });
     }
-    await exec(`helmfile deps -f ${quote(packageFileName)}`, {
+
+    const cmd: string[] = [];
+    const doc = yaml.load(newPackageFileContent) as Doc; //TODO #9610
+    const ociRepositoryRules: RepositoryRule[] = getRepositories(doc)
+      .filter(isOCIRegistry)
+      .map((value) => {
+        return {
+          ...value,
+          hostRule: hostRules.find({
+            url: 'https://' + value.url,
+            hostType: DockerDatasource.id,
+          }),
+        };
+    });
+
+    const regexOfURLPath = /\/.*/;
+    ociRepositoryRules.forEach((value) => {
+      const { username, password } = value.hostRule;
+      if (username && password) {
+        const host = value.url.replace(regexOfURLPath, '')
+
+        cmd.push(
+          `helm registry login --username ${quote(username)} --password ${quote(password)} ${host}`
+        );
+      }
+    });
```

Comment:
we should share as much code as possible with `helmv3` manager.

Reply:
yes
however, since it is outside the scope of this pull request, i do not believe i need to address it at this time.

Reply:
I agree that this should be made outside of this PR, but the refactor PR should be merged before this one.

Reply:
Okay, I will refactor in other pr.